examples/heartbleed/cheriot/heartbleed.cc

```diff
@@ -0,0 +1,356 @@
+	              ForegroundColor,
+	              Font::m5x7_16pt);
+	lcd->draw_str({5, 15},
+	              "Press Joystick to Send.",
+	              BackgroundColor,
+	              ForegroundColor,
+	              Font::m5x7_16pt);
+	lcd->draw_str({5, 30},
+	              "Request a larger buffer ",
+	              BackgroundColor,
+	              ForegroundColor,
+	              Font::m5x7_16pt);
+	lcd->draw_str({5, 40},
+	              "Suggested Length: ",
+	              BackgroundColor,
+	              ForegroundColor,
+	              Font::m5x7_16pt);
+}
+
+/**
+ * @brief Display the request length integer to the LCD.
+ *
+ * @param lcd The Sonata LCD Driver
+ * @param gpio The Sonata GPIO driver to use for I/O operations
+ * @param request_length The request length to display
+ */
+void draw_request_length(SonataLcd                *lcd,
+                         volatile SonataGpioBoard *gpio,
+                         size_t                    request_length)
+{
+	constexpr size_t ReqLenStrLen = 15u;
+	char             req_len_s[ReqLenStrLen];
+	size_t_to_str_base10(req_len_s, request_length);
+	lcd->draw_str(
+	  {110, 40}, req_len_s, BackgroundColor, ForegroundColor, Font::m5x7_16pt);
+}
+
+/**
+ * @brief This function contains the control logic loop to fetch joystick
+ * input related to the request length until the user presses the joystick
+ * to submit the request.
+ *
+ * @param lcd The Sonata LCD Driver
+ * @param gpio The Sonata GPIO driver to use for I/O operations
+ * @param request_length An out parameter containing the initial request
+ * length, and to which the final request length will be written.
+ */
+void get_request_length(SonataLcd                *lcd,
+                        volatile SonataGpioBoard *gpio,
+                        size_t                   *request_length)
+{
+	// Initial draw
+	draw_request_length(lcd, gpio, *request_length);
+
+	// Get user input
+	Debug::log("Waiting for user input on the joystick...");
+	bool inputSubmitted = false;
+	while (!inputSubmitted)
+	{
+		size_t prev_length = *request_length;
+		inputSubmitted     = length_joystick_control(gpio, request_length);
+		if (*request_length == prev_length)
+		{
+			continue; // Only re-draw when changed
+		}
+		lcd->draw_str({110, 40},
+		              "       ",
+		              BackgroundColor,
+		              ForegroundColor,
+		              Font::m5x7_16pt);
+		draw_request_length(lcd, gpio, *request_length);
+	}
+
+	Debug::log("Heartbeat submitted with length {}", (int)(*request_length));
+}
+
+/**
+ * @brief This function mocks the network and show the package on the lcd
+ * instead.
+ *
+ * @param lcd A handle to Sonata's LCD
+ * @param package The package to be sent.
+ * @param len The length of the package .
+ */
+void network_send(void *handle, const char *package, size_t len)
+{
+	constexpr uint32_t CharsPerLine = 29;
+	SonataLcd         *lcd          = (SonataLcd *)handle;
+
+	size_t w_boarder = 2;
```

Comment:
Nit: (6 times in this file)
```suggestion
	size_t w_border = 2;
```